lib/Sema/TypeCheckMacros.cpp

```diff
@@ -713,7 +713,6 @@ bool swift::expandFreestandingDeclarationMacro(
       return false;
     }
     decl->setDeclContext(dc);
-    TypeChecker::typeCheckDecl(decl);
```

Comment:
Yeah, this is great. The `TypeCheckDeclPrimary.cpp` fix subsumes the need for this.